scripts/check_requirements.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+import sys
+from pathlib import Path
+
+import yaml
+from packaging.version import parse as parse_version
+
+
+def parse_requirements(filepath):
+    """Parse requirements file and return a dict of package versions."""
+    versions = {}
+    with open(filepath) as f:
+        for line in f:
+            line = line.strip()
+            if line and not line.startswith("#"):
+                # Handle different requirement formats
+                if ">=" in line:
+                    pkg, version = line.split(">=")
+                elif "==" in line:
+                    pkg, version = line.split("==")
+                else:
+                    continue
+                versions[pkg.strip().lower()] = version.strip()
+    return versions
+
+
+def parse_precommit_config(filepath):
+    """Parse pre-commit config and extract ufmt repo rev and hook dependencies."""
+    with open(filepath) as f:
+        config = yaml.safe_load(f)
+
+    versions = {}
+    for repo in config["repos"]:
+        if "https://github.com/omnilib/ufmt" in repo.get("repo", ""):
+            # Get ufmt version from rev
+            rev = repo.get("rev", "")
+            if rev.startswith("v"):
+                versions["ufmt"] = rev[1:]  # Remove 'v' prefix
+
+            # Get dependency versions
+            for hook in repo["hooks"]:
+                if hook["id"] == "ufmt":
+                    for dep in hook.get("additional_dependencies", []):
+                        if "==" in dep:
```

Comment:
When would `==` not be part of this string? Let's error out if that occurs since we don't know how to properly handle it?

Reply:
as setup now and assuming `requirements-fmt.txt` will always be a lock file of sorts then it would always be there. This code would already error as written if `==` wasn't in the string as we'd get a comparison against none.